exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/Metadata.java

```diff
@@ -1851,9 +1860,81 @@ private static String relativize(String baseDir, String childPath) {
           .relativize(fullPathWithoutSchemeAndAuthority.toUri()));
       if (relativeFilePath.isAbsolute()) {
         throw new IllegalStateException(String.format("Path %s is not a subpath of %s.",
-            basePathWithoutSchemeAndAuthority.toUri().toString(), fullPathWithoutSchemeAndAuthority.toUri().toString()));
+            basePathWithoutSchemeAndAuthority.toUri().getPath(), fullPathWithoutSchemeAndAuthority.toUri().getPath()));
+      }
+      return relativeFilePath.toUri().getPath();
+    }
+  }
+
+  /**
+   * Used to identify metadata version by the deserialization "metadata_version" first property
+   * from the metadata cache file
+   */
+  public static class MetadataVersion {
+    @JsonProperty("metadata_version")
+    public String textVersion;
+
+    /**
+     * Supported metadata versions.
+     * Note: keep them synchronized with {@link ParquetTableMetadataBase} versions
+     */
+    enum Versions {
+      v1(Constants.V1),
+      v2(Constants.V2),
+      v3(Constants.V3),
+      v3_1(Constants.V3_1);
+
+      private final String version;
+
+      Versions(String version) {
+        this.version = version;
+      }
+
+      public String getVersion() {
+        return version;
+      }
+
+      public static Versions fromString(String version) {
+        for (Versions v : Versions.values()) {
+          if (v.version.equalsIgnoreCase(version)) {
+            return v;
+          }
+        }
+        return null;
+      }
+
+      public static class Constants {
+        public static final String V1 = "v1";
+        public static final String V2 = "v2";
+        public static final String V3 = "v3";
+        public static final String V3_1 = "v3_1";
+      }
+    }
+
+    /**
```

Comment:
One very handy thing to do for each version constant is to list what changed, possibly including the JIRA number for more information:

```
/**
 * Version 3.1: Changes the xyz property.
 * File names stored as relative paths.
 * See DRILL-1234.
 */
```